sdk/src/client/api/address.rs

```diff
@@ -92,78 +65,93 @@ impl<'a> GetAddressesBuilder<'a> {
-    /// Useful for bindings
-    pub fn set_options(mut self, options: GetAddressesBuilderOptions) -> Result<Self> {
-        if let Some(coin_type) = options.coin_type {
-            self = self.with_coin_type(coin_type);
-        };
-
-        if let Some(account_index) = options.account_index {
-            self = self.with_account_index(account_index);
+impl Default for GetAddressesOptions {
+    fn default() -> Self {
+        Self {
+            coin_type: SHIMMER_COIN_TYPE,
+            account_index: 0,
+            range: 0..ADDRESS_GAP_RANGE,
+            bech32_hrp: SHIMMER_TESTNET_BECH32_HRP.to_string(),
+            options: Default::default(),
         }
-
-        if let Some(range) = options.range {
-            self = self.with_range(range);
-        };
-
-        if let Some(bech32_hrp) = options.bech32_hrp {
-            self = self.with_bech32_hrp(bech32_hrp);
-        };
-
-        if let Some(options) = options.options {
-            self = self.with_options(options);
-        };
-
-        Ok(self)
     }
+}
 
-    /// Consume the builder and get a vector of public addresses bech32 encoded
-    pub async fn finish(self) -> Result<Vec<String>> {
-        let bech32_hrp = match self.bech32_hrp.clone() {
-            Some(bech32_hrp) => bech32_hrp,
-            None => match self.client {
-                Some(client) => client.get_bech32_hrp().await?,
-                None => SHIMMER_TESTNET_BECH32_HRP.to_string(),
-            },
-        };
-
-        let addresses = self
-            .secret_manager
-            .generate_addresses(self.coin_type, self.account_index, self.range, self.options)
+impl SecretManager {
+    /// Get a vector of public address strings
+    pub async fn get_addresses(
+        &self,
+        GetAddressesOptions {
+            coin_type,
+            account_index,
+            range,
+            bech32_hrp,
+            options,
+        }: GetAddressesOptions,
+    ) -> Result<Vec<String>> {
+        Ok(self
+            .generate_addresses(coin_type, account_index, range, options)
             .await?
             .into_iter()
             .map(|a| a.to_bech32(&bech32_hrp))
-            .collect();
+            .collect())
+    }
 
-        Ok(addresses)
+    /// Get a vector of EVM address strings
+    pub async fn get_evm_addresses(
+        &self,
+        GetAddressesOptions {
+            coin_type,
+            account_index,
+            range,
+            bech32_hrp,
+            options,
+        }: GetAddressesOptions,
+    ) -> Result<Vec<String>> {
+        Ok(self
+            .generate_evm_addresses(coin_type, account_index, range, options)
+            .await?
+            .into_iter()
+            .map(|a| {
+                bech32::encode(
+                    &bech32_hrp,
+                    bech32::ToBase32::to_base32(&a.as_ref()),
+                    bech32::Variant::Bech32,
+                )
+                .unwrap()
```

Comment:
I don't think we should bech32 encode them, prefix hex encoded is how it's used usually for example like in metamask